src/main/java/hudson/plugins/git/GitSCM.java

```diff
@@ -1063,6 +1064,13 @@ public void checkout(Run<?, ?> build, Launcher launcher, FilePath workspace, Tas
             throw new IOException("Could not checkout " + revToBuild.revision.getSha1String(), e);
         }
 
+        //read ´git describe´
+        try {
+            environment.put(GIT_DESCRIBE, git.describe(revToBuild.revision.getSha1String()));
+        } catch (GitException e) {
```

Comment:
Why do you need put it as env variable if you already implemented it in a token?